packages/components/src/Heading/Heading.tsx

```diff
@@ -0,0 +1,44 @@
+import { TextProps, TypographyComponent } from "../Text/Text";
+import React from "react";
+import { styled } from "twin.macro";
+
+type HeadingElement = "h1" | "h2" | "h3" | "h4" | "h5" | "h6";
+
+export type HeadingProps = TextProps & {
+  /**
+   * This prop can be used to specify which size heading should
+   * actually be rendered, in the case that you want to render an element
+   * as one heading but style it as if it were another. If both an `as` prop
+   * and a `size` prop are passed, the `as` will be used to determine the element
+   * and the `size` will be used to determine the styling.
+   */
+  as?: HeadingElement;
+};
+
+const HeadingComponent = styled(TypographyComponent).attrs({
```

Comment:
Are we getting much out of basing this off of the typography component? Wondering if it makes sense to isolate things between text and heading.

Or let me know if we're getting some useful stuff out of sharing code between them.

Reply:
it's probably over-DRYing. I can instead do:
- some util/typography file with helpers `styleFromColor` and `styleFromSize`
- Text and Heading create their own styled components, using those helpers

then the prop comments can be a bit clearer too